torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -60,11 +60,34 @@
 if TYPE_CHECKING:
     from collections import OrderedDict  # noqa: F401
 
+_TORCHDISTX_AVAIL = True
+try:
+    from torchdistx import fake, deferred_init
+except ImportError:
+    _TORCHDISTX_AVAIL = False
+
 
 FSDP_WRAPPED_MODULE = "_fsdp_wrapped_module"
 FSDP_PREFIX = FSDP_WRAPPED_MODULE + "." + FPW_MODULE + "."
 
 
+def _default_meta_device_init_fn(module):
+    """
+    Default initializer for modules initialized on the meta device.
+    """
+    # TODO: move module to device_ids here once device_id is available.
+    module.to_empty(device=torch.cuda.current_device())
```

Comment:
nit: to_empty(self.compute_device)?

Reply:
self.compute_device has actually not been computed when this is called. setting self.compute_device comes after meta init and it relies on the module being on CPU or GPU, the doc says:

```
# device for computation, if module is on GPU, use module.device;
        # if module is on CPU, use current device;
```

so basically this function is doing something similar as to the module being on CPU, which makes sense because this is an improved alternative over passing in CPU modules?

Reply:
Also, for GPU modules, self.compute_device will use the module that it is on, otherwise, it will just use current cuda device